drivers/amazonec2/amz/ec2.go

```diff
@@ -149,11 +149,23 @@ func (e *EC2) awsApiCall(v url.Values) (http.Response, error) {
 		return http.Response{}, fmt.Errorf("error creating request from client")
 	}
 	req.Header.Add("Content-type", "application/json")
-	awsauth.Sign(req, awsauth.Credentials{
-		AccessKeyID:     e.Auth.AccessKey,
-		SecretAccessKey: e.Auth.SecretKey,
-		SecurityToken:   e.Auth.SessionToken,
-	})
+
+	var creds awsauth.Credentials
+
+	if len(e.Auth.SessionToken) == 0 {
+		creds = awsauth.Credentials{
+			AccessKeyID:     e.Auth.AccessKey,
+			SecretAccessKey: e.Auth.SecretKey,
+		}
+	} else {
+		creds = awsauth.Credentials{
+			AccessKeyID:     e.Auth.AccessKey,
+			SecretAccessKey: e.Auth.SecretKey,
+			SecurityToken:   e.Auth.SessionToken,
+		}
+	}
```

Comment:
Isn't `creds.SecurityToken` going to end up as `""` when `e.Auth.SessionToken == ""` whether or not this `if` is there? Seems like you just need to change `awsauth.Sign` to `awsauth.Sign4`.